packages/@aws-cdk/aws-stepfunctions-tasks/test/emrcontainers/create-virtual-cluster-test.ts

```diff
@@ -0,0 +1,179 @@
+import '@aws-cdk/assert-internal/jest';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Stack } from '@aws-cdk/core';
+import { EmrContainersEksCreateVirtualCluster, EksClusterInput } from '../../lib/emrcontainers/create-virtual-cluster';
+
+const emrContainersVirtualClusterName = 'EMR Containers Virtual Cluster';
+let stack: Stack;
+let clusterId: string;
+/**
+ * To do for testing
+ * 1. Needs to test with(default) and without EksInfo and ContainerInfo, make sure it works without it - FINISHED
+ * 2. Needs to test ALL supported integration patterns and throw errors when needed - Finished
+ * 3. Need to finish testing for all policy statements - Finished
+ */
+
+beforeEach(() => {
+  stack = new Stack();
+  clusterId = 'test-eks';
+});
+
+test('Invoke emr-containers CreateVirtualCluster without required properties', () => {
+  // WHEN
+  const task = new EmrContainersEksCreateVirtualCluster(stack, 'Task', {
+    virtualClusterName: emrContainersVirtualClusterName,
+    eksCluster: EksClusterInput.fromTaskInput(sfn.TaskInput.fromText(clusterId)),
+    integrationPattern: sfn.IntegrationPattern.REQUEST_RESPONSE,
+  });
+
+  new sfn.StateMachine(stack, 'SM', {
+    definition: task,
+  });
+
+  // THEN
+  expect(stack.resolve(task.toStateJson())).toEqual({
+    Type: 'Task',
+    Resource: {
+      'Fn::Join': [
+        '',
+        [
+          'arn:',
+          {
+            Ref: 'AWS::Partition',
+          },
+          ':states:::emr-containers:createVirtualCluster',
+        ],
+      ],
+    },
+    End: true,
+    Parameters: {
+      Name: emrContainersVirtualClusterName,
+      ContainerProvider: {
+        Id: clusterId,
+        Type: 'EKS',
+      },
+    },
+  });
+});
+
+test('Invoke emr-containers CreateVirtualCluster with all required/non-required properties', () => {
+  // WHEN
+  const task = new EmrContainersEksCreateVirtualCluster(stack, 'Task', {
+    virtualClusterName: emrContainersVirtualClusterName,
+    eksCluster: EksClusterInput.fromTaskInput(sfn.TaskInput.fromText(clusterId)),
+    eksNamespace: 'kube-system',
```

Comment:
Avoid using `kube-system`. This is used by default for objects created by the Kubernetes system. Use `default` instead if unable to create a new namespace for testing.